src/internal/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,116 @@
+import { Role } from "../../components/Editor.tsx";
+import { useCallback } from "react";
+import { TemplateMetadata } from "../types/templateMetadata.ts";
+import { DevLogger } from "../../utils/devLogger.ts";
+
+const ROLE = "ROLE_",
+  SITE = "SITE_",
+  TEMPLATE = "TEMPLATE_",
+  LAYOUT = "LAYOUT_",
+  ENTITY = "ENTITY_",
+  THEME = "THEME_";
+
+const devLogger = new DevLogger();
+
+/**
+ * useLocalStorage contains helper functions for reading and clearing local storage
+ * @param templateMetadata The template metadata for the active template (undefined if not loaded)
+ */
+export const useLocalStorage = (
+  templateMetadata: TemplateMetadata | undefined
+) => {
+  const buildVisualConfigLocalStorageKey = useCallback(() => {
+    if (!templateMetadata) {
+      return "";
+    }
+
+    return getVisualConfigLocalStorageKey(
+      templateMetadata.isDevMode && !templateMetadata.devOverride,
+      templateMetadata.role,
+      templateMetadata.templateId,
+      templateMetadata.layoutId,
+      templateMetadata.entityId
+    );
+  }, [templateMetadata]);
+
+  const buildThemeLocalStorageKey = useCallback(() => {
+    if (!templateMetadata) {
+      return "";
+    }
+
+    return getThemeLocalStorageKey(
+      templateMetadata.isDevMode,
+      templateMetadata.siteId,
+      templateMetadata.themeEntityId
+    );
+  }, [templateMetadata]);
+
+  const clearLocalStorage = () => {
```

Comment:
no need to change for this PR, but I feel like having these two be linked is bad

Reply:
Good call, will fix in next PR